File: _legacy-swift/Services/AnalyticsService.swift
import Foundation
import os

/// Analytics event names. Add cases conservatively; every event is considered
/// part of the analytics contract and tracked in CI.
public enum AnalyticsEvent: String, Sendable {
    case appOpen = "app_open"
    case videoSelected = "video_selected"
    case exportStandardStarted = "export_standard_started"
    case rewardedTrialRequested = "rewarded_trial_requested"
    case rewardedTrialCompleted = "rewarded_trial_completed"
    case rewardedTrialFailed = "rewarded_trial_failed"
    case exportHQStarted = "export_hq_started"
    case exportCompleted = "export_completed"
    case exportFailed = "export_failed"
    case premiumPaywallViewed = "premium_paywall_viewed"
    case premiumPurchased = "premium_purchased"
    case restorePurchaseTapped = "restore_purchase_tapped"
}

/// Protocol-based analytics abstraction. Implementations must be side-effect
/// free in tests.
public protocol AnalyticsServiceProtocol: Sendable {
    func track(_ event: AnalyticsEvent)
    func track(_ event: AnalyticsEvent, parameters: [String: String])
}

/// Default implementation that logs via `os.Logger` in Debug builds only.
public final class AnalyticsService: AnalyticsServiceProtocol {
    private let logger = Logger(subsystem: "jp.example.livephotomaker", category: "analytics")

    public init() {}

    public func track(_ event: AnalyticsEvent) {
        track(event, parameters: [:])
    }

    public func track(_ event: AnalyticsEvent, parameters: [String: String]) {
        #if DEBUG
        if parameters.isEmpty {
            logger.debug("event=\(event.rawValue, privacy: .public)")
        } else {
            logger.debug("event=\(event.rawValue, privacy: .public) params=\(parameters, privacy: .public)")
        }
        #endif
        // TODO: forward to a production analytics backend when integrated.
    }
}

File: src/services/AnalyticsService.ts
import { logger } from '@/utils/logger';

export type AnalyticsEvent =
  | 'app_open'
  | 'video_selected'
  | 'export_standard_started'
  | 'rewarded_trial_requested'
  | 'rewarded_trial_completed'
  | 'rewarded_trial_failed'
  | 'export_hq_started'
  | 'export_completed'
  | 'export_failed'
  | 'premium_paywall_viewed'
  | 'premium_purchased'
  | 'restore_purchase_tapped';

export interface AnalyticsService {
  track(event: AnalyticsEvent, params?: Record<string, string>): void;
}

export const createAnalyticsService = (): AnalyticsService => ({
  track: (event, params) => {
    logger.debug('analytics', event, params ?? {});
    // Real implementation would forward to an analytics backend here.
  },
});
